### ISSUE TITLE
Possibility to style Add to cart with options block

### ISSUE THREAD

**Issue by tomxygen** [woothemes/woocommerce#42356]
## Is your feature request related to a problem? Please describe.
Currently, it's not possible to apply any kind of styling to the add to cart with options block.
It takes the styling of the button block and it's not possible to apply custom styling like a custom color.

## Describe the solution you'd like
I'm creating different product templates that are applied to different products, and I'd like to have different Add to cart button colors.

**Comment by gigitux** [comment]
Hi @tomxygen,
Thanks for opening this issue!

> It takes the styling of the button block and it's not possible to apply custom styling like a custom color.

Did you try with [the Additional Classes](https://wordpress.com/support/wordpress-editor/adding-additional-css-classes-to-blocks/) and see if you can customize the block?

Additionally, we acknowledge that it would be more convenient to style the block using block style settings. While we plan to enhance several blocks in the future, we cannot provide a specific timeline for these improvements at the moment.

I would like to take this opportunity to invite you to share your valuable feedback on our roadmap. Please feel free to add a comment to [ this post ](https://developer.woocommerce.com/2023/03/24/store-editing-roadmap-update-q1-2023/)to help shape our work going forward. We greatly appreciate your input and look forward to hearing your thoughts.

**Comment by tomxygen** [comment]
Hey @gigitux 
Thanks for your suggestion.
I tried Additional classes, but unfortunately the class is applied to the entire block and not to the button.
This is what happens when I add a background color to the additional class.
<img width="562" alt="Screenshot 2023-05-31 at 11 55 35 AM" src="https://github.com/woocommerce/woocommerce-blocks/assets/47855768/6e6332e2-361c-4a6b-b960-5fe41586cb0f">

**Comment by gigitux** [comment]
You are right! The additional classes are applied to the block container. Since the template is within a Group block we can use Additional Classes on it (to distinguish which template is loaded). After with additional styles, you can target the Add To Cart button. This is the final result:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/78125f08-0c6f-4765-8814-a882a8956cee)

This is the CSS that I used:

```
.album.wp-block-add-to-cart-form button {
    background: red;
}
```

![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/93f9c0e9-2c3a-453f-8de8-578002aa7116)

![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/fab22474-dc7c-4fbb-a78d-fbb7f47b54e3)

**Comment by tomxygen** [comment]
Hi @gigitux thanks for your reply and suggestions, I’ll definitely implement them until a better way to style the ATC button is implemented.

But again, I really think that it should be implemented. Almost all blocks do have that kind of customization, i think the ATC with options block should have them too.
It’s especially important for me because I’m listing a few products on my store and I’d like to have their ATC button color matched with the main color of the product, so I’m creating a product template for each product in order to style all the colors of that page.

**Comment by gigitux** [comment]
> Hi @gigitux thanks for your reply and suggestions, I’ll definitely implement them until a better way to style the ATC button is implemented.
> 
> But again, I really think that it should be implemented. Almost all blocks do have that kind of customization, i think the ATC with options block should have them too. It’s especially important for me because I’m listing a few products on my store and I’d like to have their ATC button color matched with the main color of the product, so I’m creating a product template for each product in order to style all the colors of that page.

As I said above, we acknowledge that it would be more convenient to style the block using block style settings, but we cannot provide a specific timeline for these improvements at the moment.

Take a look at woocommerce/woocommerce-blocks#9686, it could help you. Instead to add the class manually to the group block, the product-specific classes will be added in the body and it will be easier to target the specific dom elements via CSS.

**Comment by asafm7** [comment]
@gigitux 

> It takes the styling of the button block and it's not possible to apply custom styling like a custom color.

Actually, that would have been great. But it isn't the case for me. The global button style doesn't affect the "Add to Cart with Options" button.

This is the selector for the global button style: `.wp-block-button .wp-block-button__link`.

It doesn't match the "Add to Cart with Options" button.

As the traditional Customize menu is now hidden, I think it leaves no way to change the buttons' style without a custom code. If true, it feels like a substantial gap for an e-commerce platform.

Am I missing something?

**Comment by gigitux** [comment]
Hi @asafm7,
As I said above, we acknowledge that it would be more convenient to style the block using block style settings, but we cannot provide a specific timeline for these improvements at the moment.

**Comment by asafm7** [comment]
Thank you @gigitux .

The issue opened with the statement that it is _possible_ to style the add to cart button using the global button block styling, but not individually. While not perfect, it is something.

I wanted to stress that this is inaccurate - it isn't even possible to style the add-to-cart button using the global button block styling - and there is no way to affect the style without custom coding. I thought this fact might put the gap in a different light, and change its priority.

It is actually not about adding the ability to style the add to cart button, but about giving it (and its ancestors) the correct CSS classes to follow the global button styling.

Thank you again for your help and work.

**Comment by kkoteen** [comment]
Hi @tomxygen - thanks for reporting this, and for the detail you gave.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

It looks like you can now customize the "add to cart" button in the global settings. 

I'm going to close this one. If you're still facing this issue on a current version or I've misunderstood, we'll happily take another look.
